e2e/config/vhd.go

```diff
@@ -0,0 +1,176 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+)
+
+// VHD represents a VHD used to run AgentBaker E2E scenarios.
+type VHD struct {
+	ImageID         string
+	VersionTagName  string
+	VersionTagValue string
+	// ResourceID is the resource ID pointing to the underlying VHD in Azure. Based on the current setup, this will always be the resource ID
+	// of an image version in a shared image gallery.
+	resourceID               VHDResourceID
+	buildResourceID          VHDResourceID
+	fetchBuildResourceIDOnce sync.Once
+	fetchResourceIDOnce      sync.Once
+}
+
+var (
+	VHDUbuntu1804Gen2Containerd = &VHD{
+		ImageID:         "/subscriptions/8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8/resourceGroups/aksvhdtestbuildrg/providers/Microsoft.Compute/galleries/PackerSigGalleryEastUS/images/1804Gen2",
+		VersionTagName:  "branch",
+		VersionTagValue: "refs/heads/master",
+	}
+	VHDUbuntu2204Gen2Arm64Containerd = &VHD{
+		ImageID:         "/subscriptions/8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8/resourceGroups/aksvhdtestbuildrg/providers/Microsoft.Compute/galleries/PackerSigGalleryEastUS/images/2204Gen2Arm64",
+		VersionTagName:  "branch",
```

Comment:
could you add a description as to what the expectation of finding a tag with name `branch` would mean ?